radicle-httpd/src/api/json.rs

```diff
@@ -46,14 +47,18 @@ pub(crate) fn session(session_id: String, session: &Session) -> Value {
 }
 
 /// Returns JSON for a blob with a given `path`.
-pub(crate) fn blob(blob: &Blob, path: &str) -> Value {
-    json!({
-        "binary": blob.is_binary(),
-        "content": blob.content(),
-        "name": name_in_path(path),
-        "path": path,
-        "lastCommit": commit(blob.commit())
-    })
+pub(crate) fn blob<T: AsRef<[u8]>>(blob: &Blob<T>, path: &str) -> Value {
+    if let Ok(content) = str::from_utf8(blob.content()) {
+        json!({
```

Comment:
`blob` is now generic and returns a vec of bytes, thus I'm converting it into a utf8 string here. do we want to treat binary blobs differently?

Reply:
Good question. The previous version relies on `BlobContent` type serialization provided by `radicle-surf` itself.  Now we removed `BlobContent` and return `&[u8]` directly. 

Your change looks good to me, and you can add handling for the binary case too, for example converting into Base64. That provides the max freedom and not to rely on the serialization from `radicle-surf`. 

That said, I think we can add a couple of new methods in `radicle-surf` to help: 

`blob.content_str()` that returns `&str` when the content is not binary, otherwise returns empty.  
`blob.content_base64()` that returns `String` encoded in Base64. The caller uses this method when `is_binary()` is true.

(The reason to have two methods rather than one method is that, I don't want to clone a new string in the regular case of text content).

Reply:
I thought about it a bit more. I still recommend to handle the serialization of both plain and binary case in this function. Something like this: 

```rust
    let base64_string: String;
    let content = match str::from_utf8(blob.content()) {
        Ok(s) => s,
        Err(_) => { base64_string = base64::encode(blob.content()); &base64_string }
    };
    json!({
        "binary": blob.is_binary(),
        "content": content,
        "name": name_in_path(path),
        "path": path,
        "lastCommit": commit(blob.commit())
    })
```

I think adding `blob.content_str()` and `blob.content_base64()` in `radicle-surf` is against our basic idea of letting our client (i.e. `httpd`) to do their serialization. And such methods' purposes are too narrow, not something coherent with the blob itself.

Reply:
not sure about the benefits of using `base64`, doesn't github also return string representation of binary files when clicking raw?

Reply:
also can `str::from_utf8_unchecked` be used here?

Reply:
I wouldn't do that unless we're 100% sure that all non-binary files are UTF-8.